pkg/core/interop/contract/call.go

```diff
@@ -99,3 +98,19 @@ func CallExInternal(ic *interop.Context, cs *state.Contract,
 
 	return nil
 }
+
+// CallFromNative performs synchronous call from native contract.
+func CallFromNative(ic *interop.Context, cs *state.Contract, method string, args []stackitem.Item, checkReturn vm.CheckReturnState) error {
+	startSize := ic.VM.Istack().Len()
+	if err := CallExInternal(ic, cs, method, args, smartcontract.All, checkReturn); err != nil {
+		return err
+	}
+
+	for !ic.VM.HasStopped() && ic.VM.Istack().Len() > startSize {
+		_ = ic.VM.Step()
```

Comment:
This implies that VM enters `FaultState` if there is an error returned here. While this is always true now, I'd rather still check for error here and return it (it also allows to be more specific about error happened).

Reply:
Fixed.